### PR TITLE
Iteration 1 con

### PR DESCRIPTION
**Pull request by cmougios-ITC205**
please review code. just updated a few minutes ago

### CODE REVIEW

File: BorrowBookControl.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class BorrowBookControl {
+
+	private BorrowBookUI ui;
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT Custom: ui should always be UI (As discussed), give this member a more meaningful name. bookUI is fine.

File: BorrowBookControl.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	private enum ControlState { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
+	private ControlState state;
+	private List<book> pending;
+	private List<loan> completed;
+	private book book;
+
+
+	public BorrowBookControl() {
+		this.library = library.INSTANCE();
+		state = ControlState.INITIALISED;
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	}
+
+
+	public void setUI(BorrowBookUI ui) {
+		if (!state.equals(ControlState.INITIALISED))
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline 7: There MUST be a SINGLE consistent bracketing style used throughout the code

File: BorrowBookControl.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+			throw new RuntimeException("BorrowBookControl: cannot call setUI except in INITIALISED state");
+
+		this.ui = ui;
+		ui.setState(BorrowBookUI.UIState.READY);
+		state = ControlState.READY;
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+	}
+
+
+	public void swiped(int memberId) {
+		if (!state.equals(ControlState.READY))
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline 7: There MUST be a SINGLE consistent bracketing style used throughout the code

File: BorrowBookControl.java [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+			throw new RuntimeException("BorrowBookControl: cannot call cardSwiped except in READY state");
+
+		member = library.MEMBER(memberId);
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+		if (member == null) {
+			ui.display("Invalid memberId");
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+			return;
+		}
+		if (library.MEMBER_CAN_BORROW(member)) {
+			pending = new ArrayList<>();
+			ui.setState(BorrowBookUI.UIState.SCANNING);
+			state = ControlState.SCANNING; }
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+		if (!state.equals(ControlState.READY))
+			throw new RuntimeException("BorrowBookControl: cannot call cardSwiped except in READY state");
+
+		member = library.MEMBER(memberId);
+		if (member == null) {
+			ui.display("Invalid memberId");
+			return;
+		}
+		if (library.MEMBER_CAN_BORROW(member)) {
+			pending = new ArrayList<>();
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+			ui.setState(BorrowBookUI.UIState.SCANNING);
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java (hunk 8 above)
Comment (nullreferrer) [reply BorrowBookControl.java]: DEFECT: Guideline 7: There MUST be a SINGLE consistent bracketing style used throughout the code
> if (condition) {
> // Code
> } else {
> // Other Code
> }

File: BorrowBookControl.java [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+			state = ControlState.SCANNING; }
+		else {
+			ui.display("Member cannot borrow at this time");
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+			ui.setState(BorrowBookUI.UIState.RESTRICTED);
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+		}
+	}
+
+
+	public void scanned(int bookId) {
+		book = null;
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+		if (!state.equals(ControlState.SCANNING)) {
+			throw new RuntimeException("BorrowBookControl: cannot call bookScanned except in SCANNING state");
+		}
+		book = library.Book(bookId);
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+		else {
+			ui.display("Member cannot borrow at this time");
+			ui.setState(BorrowBookUI.UIState.RESTRICTED);
+		}
+	}
+
+
+	public void scanned(int bookId) {
+		book = null;
+		if (!state.equals(ControlState.SCANNING)) {
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+			throw new RuntimeException("BorrowBookControl: cannot call bookScanned except in SCANNING state");
+		}
+		book = library.Book(bookId);
+		if (book == null) {
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+			ui.display("Invalid bookId");
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+			return;
+		}
+		if (!book.AVAILABLE()) {
+			ui.display("Book cannot be borrowed");
+			return;
+		}
+		pending.add(book);
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+		if (!state.equals(ControlState.SCANNING)) {
+			throw new RuntimeException("BorrowBookControl: cannot call bookScanned except in SCANNING state");
+		}
+		book = library.Book(bookId);
+		if (book == null) {
+			ui.display("Invalid bookId");
+			return;
+		}
+		if (!book.AVAILABLE()) {
+			ui.display("Book cannot be borrowed");
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 20]
```diff
@@ -0,0 +1,116 @@
...
+			return;
+		}
+		pending.add(book);
+		for (book addBook : pending) {
+			ui.display(addBook.toString());
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

Comment (nullreferrer) [reply BorrowBookControl.java]: DEFECT: Guideline 9: The argument list of any method calls MUST NOT contain any other method calls “toString()”

File: BorrowBookControl.java [hunk 21]
```diff
@@ -0,0 +1,116 @@
...
+		}
+		if (library.Loans_Remaining_For_Member(member) - pending.size() == 0) {
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier x2

File: BorrowBookControl.java [hunk 22]
```diff
@@ -0,0 +1,116 @@
...
+			ui.display("Loan limit reached");
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 23]
```diff
@@ -0,0 +1,116 @@
...
+			complete();
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 24]
```diff
@@ -0,0 +1,116 @@
...
+		}
+	}
+
+
+	public void complete() {
+		if (pending.size() == 0) {
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 25]
```diff
@@ -0,0 +1,116 @@
...
+			cancel();
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 26]
```diff
@@ -0,0 +1,116 @@
...
+		}
+		else {
+			ui.display("\nFinal Borrowing List");
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 27]
```diff
@@ -0,0 +1,116 @@
...
+			for (book displayBook : pending) {
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 28]
```diff
@@ -0,0 +1,116 @@
...
+				ui.display(displayBook.toString());
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 29]
```diff
@@ -0,0 +1,116 @@
...
+			}
+			completed = new ArrayList<loan>();
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 30]
```diff
@@ -0,0 +1,116 @@
...
+			ui.setState(BorrowBookUI.UIState.FINALISING);
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 31]
```diff
@@ -0,0 +1,116 @@
...
+			state = ControlState.FINALISING;
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 32]
```diff
@@ -0,0 +1,116 @@
...
+			 }
```
Comment (nullreferrer) [inline BorrowBookControl.java]: Something weird here, the indenting is all bizzare. Just check that the file is all using 4 spaces as tabs and that there is no mix of the 2.

File: BorrowBookControl.java [hunk 33]
```diff
@@ -0,0 +1,116 @@
...
+	}
+
+
+	public void commitLoans() {
+		if (!state.equals(ControlState.FINALISING)) {
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 34]
```diff
@@ -0,0 +1,116 @@
...
+			throw new RuntimeException("BorrowBookControl: cannot call commitLoans except in FINALISING state");
+		}
+		for (book loanBook : pending) {
+			loan LOAN = library.ISSUE_LAON(loanBook, member);
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier x2

File: BorrowBookControl.java [hunk 35]
```diff
@@ -0,0 +1,116 @@
...
+			state = ControlState.FINALISING;
+			 }
+	}
+
+
+	public void commitLoans() {
+		if (!state.equals(ControlState.FINALISING)) {
+			throw new RuntimeException("BorrowBookControl: cannot call commitLoans except in FINALISING state");
+		}
+		for (book loanBook : pending) {
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 36]
```diff
@@ -0,0 +1,116 @@
...
+			loan LOAN = library.ISSUE_LAON(loanBook, member);
+			completed.add(LOAN);
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 37]
```diff
@@ -0,0 +1,116 @@
...
+		}
+		ui.display("Completed Loan Slip");
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 38]
```diff
@@ -0,0 +1,116 @@
...
+		for (loan LOAN : completed) {
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 39]
```diff
@@ -0,0 +1,116 @@
...
+			ui.display(LOAN.toString());
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 40]
```diff
@@ -0,0 +1,116 @@
...
+		}
+		ui.setState(BorrowBookUI.UIState.COMPLETED);
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 41]
```diff
@@ -0,0 +1,116 @@
...
+		state = ControlState.COMPLETED;
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 42]
```diff
@@ -0,0 +1,116 @@
...
+	}
+
+
+	public void cancel() {
+		ui.setState(BorrowBookUI.UIState.CANCELLED);
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 43]
```diff
@@ -0,0 +1,116 @@
...
+		state = ControlState.CANCELLED;
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookUI.java [hunk 44]
```diff
@@ -0,0 +1,128 @@
...
+	public static enum UIState { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
+
+	private BorrowBookControl control;
+	private Scanner input;
+	private UIState state;
+
+	
+	public BorrowBookUI(BorrowBookControl control) {
+		this.control = control;
+		input = new Scanner(System.in);
```
Comment (nullreferrer) [inline BorrowBookUI.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookUI.java [hunk 45]
```diff
@@ -0,0 +1,128 @@
...
+		this.state = UIState.INITIALISED;
+		this.control.setUI(this);
+		/*control.setUI(this);*/
+	}
+
+	
+	private String input(String prompt) {
+		/*System.out.print(prompt);*/
```
Comment (nullreferrer) [inline BorrowBookUI.java]: Another weird indenting issue. Just check this same file is using 4 spaces as tabs and that there are no tabs in the file

File: BorrowBookUI.java [hunk 46]
```diff
@@ -0,0 +1,128 @@
...
+        System.out.print(this);
+		return input.nextLine();
```
Comment (nullreferrer) [inline BorrowBookUI.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookUI.java [hunk 47]
```diff
@@ -0,0 +1,128 @@
...
+	}
+
+	
+	public void run() {
+
+		output("Borrow Book Use Case UI\n");
+		
+		while (true) {
+			
+			switch (state) {
```
Comment (nullreferrer) [inline BorrowBookUI.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookUI.java [hunk 48]
```diff
@@ -0,0 +1,128 @@
...
+			
+			case CANCELLED:
+				output("Borrowing Cancelled");
```
Comment (nullreferrer) [inline BorrowBookUI.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookUI.java [hunk 49]
```diff
@@ -0,0 +1,128 @@
...
+				return;
+
+				
+			case READY:
+				String memberCard = input("Swipe member card (press <enter> to cancel): ");
```
Comment (nullreferrer) [inline BorrowBookUI.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookUI.java [hunk 50]
```diff
@@ -0,0 +1,128 @@
...
+				}
+				catch (NumberFormatException e) {
+
+					output("Invalid Member Id");
+				}
+				break;
+
+				
+			case RESTRICTED:
+				input("Press <any key> to cancel");
```
Comment (nullreferrer) [inline BorrowBookUI.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookUI.java [hunk 51]
```diff
@@ -0,0 +1,128 @@
...
+				this.control.cancel();
+				break;
+			
+				
+			case SCANNING:
+				String scanBook = input("Scan Book (<enter> completes): ");
```
Comment (nullreferrer) [inline BorrowBookUI.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookUI.java [hunk 52]
```diff
@@ -0,0 +1,128 @@
...
+				if (scanBook.length() == 0) {
+
+					this.control.complete();
+					break;
+				}
+				try {
+					int bookId = Integer.valueOf(scanBook).intValue();
+					control.scanned(bookId);
```
Comment (nullreferrer) [inline BorrowBookUI.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookUI.java [hunk 53]
```diff
@@ -0,0 +1,128 @@
...
+					
+				} catch (NumberFormatException e) {
+					output("Invalid Book Id");
```
Comment (nullreferrer) [inline BorrowBookUI.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookUI.java [hunk 54]
```diff
@@ -0,0 +1,128 @@
...
+				} 
+				break;
+					
+				
+			case FINALISING:
+				String askQuestion = input("Commit loans? (Y/N): ");
```
Comment (nullreferrer) [inline BorrowBookUI.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookUI.java [hunk 55]
```diff
@@ -0,0 +1,128 @@
...
+				if (askQuestion.toUpperCase().equals("N")) {
+					this.control.cancel();
+					
+				} else {
+					control.commitLoans();
+					input("Press <any key> to complete ");
```
Comment (nullreferrer) [inline BorrowBookUI.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookUI.java [hunk 56]
```diff
@@ -0,0 +1,128 @@
...
+				}
+				break;
+				
+				
+			case COMPLETED:
+				output("Borrowing Completed");
```
Comment (nullreferrer) [inline BorrowBookUI.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookUI.java [hunk 57]
```diff
@@ -0,0 +1,128 @@
...
+				return;
+	
+				
+			default:
+				output("Unhandled state");
```
Comment (nullreferrer) [inline BorrowBookUI.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

**Review by nullreferrer — CHANGES_REQUESTED** [review]
Please review the items I have mentioned.

File: BorrowBookControl.java (hunk 1 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: done

File: BorrowBookControl.java (hunk 2 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: done

File: BorrowBookControl.java (hunk 3 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: done

File: BorrowBookControl.java (hunk 4 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: done

File: BorrowBookControl.java (hunk 5 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: done

File: BorrowBookControl.java (hunk 6 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: done

Comment (cmougios-ITC205) [reply BorrowBookControl.java]: done

File: BorrowBookControl.java (hunk 7 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: done

File: BorrowBookUI.java (hunk 53 above)
Comment (cmougios-ITC205) [reply BorrowBookUI.java]: done

File: BorrowBookUI.java (hunk 54 above)
Comment (cmougios-ITC205) [reply BorrowBookUI.java]: udpated

File: BorrowBookUI.java (hunk 55 above)
Comment (cmougios-ITC205) [reply BorrowBookUI.java]: done

File: BorrowBookUI.java (hunk 56 above)
Comment (cmougios-ITC205) [reply BorrowBookUI.java]: updated

File: BorrowBookUI.java (hunk 57 above)
Comment (cmougios-ITC205) [reply BorrowBookUI.java]: updated

File: BorrowBookControl.java [hunk 58]
```diff
@@ -0,0 +1,117 @@
...
+
+
+	public BorrowBookControl() {
+		this.library = library.INSTANCE();
+		this.state = ControlState.INITIALISED;
+	}
+
+
+	public void setUI(BorrowBookUI bookUI) {
+		if (!state.equals(ControlState.INITIALISED)) {
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 59]
```diff
@@ -0,0 +1,117 @@
...
+			throw new RuntimeException("BorrowBookControl: cannot call setUI except in INITIALISED state");
+		}
+		this.bookUI = bookUI;
+		bookUI.setState(BorrowBookUI.UIState.READY);
+		this.state = ControlState.READY;
+	}
+
+
+	public void swiped(int memberId) {
+		if (!state.equals(ControlState.READY)) {
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java (hunk 9 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: added this qualifier to bookUI

File: BorrowBookControl.java (hunk 13 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: updated

File: BorrowBookControl.java (hunk 15 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: updated

File: BorrowBookControl.java (hunk 14 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: this has already been added from last push. Not sure what's going on!
 this.book = library.Book(bookId);

File: BorrowBookControl.java (hunk 16 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: done

File: BorrowBookControl.java (hunk 18 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: updated

File: BorrowBookControl.java (hunk 21 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: updated

File: BorrowBookControl.java (hunk 23 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: updated

File: BorrowBookControl.java (hunk 24 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: updated

File: BorrowBookControl.java (hunk 27 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: updated

File: BorrowBookControl.java (hunk 29 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: updated

File: BorrowBookControl.java (hunk 31 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: updated

File: BorrowBookControl.java (hunk 32 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: updated, removed tabs and included 4 spaces as tabs

File: BorrowBookControl.java (hunk 34 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: updated. Not sure if i did this right though. do you mind checking? thanks

File: BorrowBookControl.java (hunk 36 above)
Comment (cmougios-ITC205) [reply BorrowBookControl.java]: updated

File: BorrowBookControl.java [hunk 60]
```diff
@@ -0,0 +1,168 @@
...
+
+    this.library = library.INSTANCE();
+    this.state = ControlState.INITIALISED;
+
+	}
+
+
+	public void setUI(BorrowBookUI bookUI) {
+
+    if (!state.equals(ControlState.INITIALISED)) {
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 61]
```diff
@@ -0,0 +1,168 @@
...
+    this.bookUI = bookUI;
+    this.bookUI.setState(BorrowBookUI.UIState.READY);
+    this.state = ControlState.READY;
+
+	}
+
+
+	public void swiped(int memberId) {
+
+    if (!state.equals(ControlState.READY)) {
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 62]
```diff
@@ -0,0 +1,168 @@
...
+    this.member = library.MEMBER(memberId);
+
+    if (this.member == null) {
+
+	this.bookUI.display("Invalid memberId");
+	return;
+
+	}
+
+    if (library.MEMBER_CAN_BORROW(member)) {
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 63]
```diff
@@ -0,0 +1,168 @@
...
+
+	pending = new ArrayList<>();
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 64]
```diff
@@ -0,0 +1,168 @@
...
+	this.bookUI.setState(BorrowBookUI.UIState.RESTRICTED);
+
+	}
+	}
+
+
+	public void scanned(int bookId) {
+
+    this.book = null;
+    if (!state.equals(ControlState.SCANNING)) {
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 65]
```diff
@@ -0,0 +1,168 @@
...
+	return;
+
+	}
+	if (!book.AVAILABLE()) {
+
+	this.bookUI.display("Book cannot be borrowed");
+	return;
+
+	}
+	pending.add(book);
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 66]
```diff
@@ -0,0 +1,168 @@
...
+	}
+
+	this.book = library.Book(bookId);
+	if (this.book == null) {
+
+	this.bookUI.display("Invalid bookId");
+	return;
+
+	}
+	if (!book.AVAILABLE()) {
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 67]
```diff
@@ -0,0 +1,168 @@
...
+
+	this.bookUI.display("Book cannot be borrowed");
+	return;
+
+	}
+	pending.add(book);
+	for (book addBook : pending) {
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier (pending)

File: BorrowBookControl.java [hunk 68]
```diff
@@ -0,0 +1,168 @@
...
+
+	this.bookUI.display(addBook.toString());
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline 9: The argument list of any method calls MUST NOT contain any other method calls “toString()”

File: BorrowBookControl.java [hunk 69]
```diff
@@ -0,0 +1,168 @@
...
+
+	}
+	if (library.Loans_Remaining_For_Member(member) - pending.size() == 0) {
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier x3

File: BorrowBookControl.java [hunk 70]
```diff
@@ -0,0 +1,168 @@
...
+	this.bookUI.display("Loan limit reached");
+	this.complete();
+
+	}
+	}
+
+
+	public void complete() {
+
+    if (pending.size() == 0) {
```
Comment (nullreferrer) [inline BorrowBookControl.java:112]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier

File: BorrowBookControl.java [hunk 71]
```diff
@@ -0,0 +1,168 @@
...
+
+	public void commitLoans() {
+
+    if (!this.state.equals(ControlState.FINALISING)) {
+
+    throw new RuntimeException("BorrowBookControl: cannot call commitLoans except in FINALISING state");
+
+    }
+
+    for (book loanBook : pending) {
```
Comment (nullreferrer) [inline BorrowBookControl.java:141]: DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier (pending)

File: BorrowBookControl.java [hunk 72]
```diff
@@ -0,0 +1,168 @@
...
+
+	loan LOAN = this.library.ISSUE_LAON(loanBook, member);
```
Comment (nullreferrer) [inline BorrowBookControl.java]: DEFECT: Guideline 1: All variable names must be meaningful and are to start with a lowercase letter and to be in camelBack.

**Review by nullreferrer — CHANGES_REQUESTED** [review]
See comments

**Comment by nullreferrer** [comment]
Incomplete work. Moving to iteration 2 and abandoning feature branch.

**Pull request closed without merging**